manim/utils/tex.py

```diff
@@ -1,298 +1,203 @@
+            else TexTemplate.default_documentclass
+        )
+        self.preamble = (
+            preamble if preamble is not None else TexTemplate.default_preamble
+        )
+        self.placeholder_text = (
+            placeholder_text
+            if placeholder_text is not None
+            else TexTemplate.default_placeholder_text
+        )
+        self.post_doc_commands = (
+            post_doc_commands
+            if post_doc_commands is not None
+            else TexTemplate.default_post_doc_commands
+        )
+        self.rebuild()
+
+    def rebuild(self):
+        self.body = (
+            self.documentclass
+            + "\n"
+            + self.preamble
+            + "\n"
+            + r"\begin{document}"
+            + "\n"
+            + self.post_doc_commands
+            + "\n"
+            + self.placeholder_text
+            + "\n"
+            + r"\end{document}"
+            + "\n"
         )
-        for pkg in self.common_packages:
-            tpl += self.generate_usepackage(pkg)
-
-        if self.use_ctex:
-            for pkg in self.ctex_packages:
-                tpl += self.generate_usepackage(pkg)
-        else:
-            for pkg in self.tex_packages:
-                tpl += self.generate_usepackage(pkg)
-
-        tpl += self.common_preamble_text
-        if self.use_ctex:
-            tpl += self.ctex_preamble_text
-        else:
-            tpl += self.tex_preamble_text
-
-        tpl += "\n" r"\begin{document}" "\n"
-        tpl += f"\n{self.text_to_replace}\n"
-        tpl += "\n" r"\end{document}"
-
-        self.body = tpl
-
-    def prepend_package(self, pkg):
-        """Adds a new package (or several new packages)
-        before all other packages. Sometimes, the order of
-        the ``\\usepackage`` directives is relevant.
-
-        Parameters
-        ----------
-        pkg : :class:`str`
-            The package name, e.g. ``"siunitx"``
-        """
-        self.common_packages.insert(0, pkg)
-        self.rebuild_cache()
 
-    def append_package(self, pkg):
-        """Adds a new package (or several new packages)
-        after all other packages. Sometimes, the order of
-        the ``\\usepackage`` directives is relevant.
+    def prepend_to_preamble(self, txt):
+        """Adds txt to the TeX template preamble just after the \documentclass
 
         Parameters
         ----------
-        pkg : :class:`str`
-            The package name, e.g. ``"siunitx"``
+        txt : :class:`str`
+            String containing the text to be added, e.g. ``\\usepackage{siunitx}``
         """
-        self.common_packages.append(pkg)
-        self.rebuild_cache()
+        self.preamble = txt + "\n" + self.preamble
+        self.rebuild()
 
-    def append_to_preamble(self, text):
-        """Adds commands (e.g. macro definitions) at the end of the preamble.
+    def add_to_preamble(self, txt):
+        """Adds txt to the TeX template preamble just before \\begin{document}
```

Comment:
```suggestion
    def append_to_preamble(self, txt):
        """Adds stuff (e.g. definitions, packages) to the TeX template preamble just before \\begin{document}
```

The other part is called `prepend`, so I guess it would be good to use `append` here.

Reply:
I though about that, but then I thought adding a package to the preamble is something that may get used once in a while, while prepending to the preamble is really a very special use function with only the rarest of uses.
add to preamble and add to document are thus the generic cases and prepend is special. no?

Reply:
.... I'm not strongly opposed or anything, but if you change it here, you have to change it in the example_scenes in the tex_templates and in the examples in the documentation.

Reply:
I am fine with both of them. I see an advantage in the sense that `append` indicates (without looking at the docs) where the code is added. This can be important in some cases, e.g. `hyperref` -- which is of course a bad example, because we probably do not use it in a manimation.

Reply:
> if you change it here, you have to change it in the example_scenes in the tex_templates and in the examples in the documentation.

Absolutely. I didn't want to suggest that as long as it was not clear whether the name change was deemed useful. :-)

Reply:
any solution is fine by me including add_to_preamble(txt, prepend=true) if you feel so inclined :D
knock yourself out!

but tbh saying "using `add_to_preamble` is how you add an extra latex packed" works 99% of the time so...